src/acrn/acrn_driver.c

```diff
@@ -2593,8 +2595,14 @@ acrnStateInitialize(bool privileged,
     if (virCapabilitiesGetNodeInfo(&acrn_driver->nodeInfo) < 0)
         goto cleanup;
 
-    if (acrnInitPlatform(&acrn_driver->pi, &acrn_driver->nodeInfo,
-                         &acrn_driver->vcpuAllocMap) < 0)
+    ret = acrnInitPlatform(&acrn_driver->pi, &acrn_driver->nodeInfo,
+        &acrn_driver->vcpuAllocMap);
+    if (ret == -ENODEV) {
+        /* we are not running on an ACRN enabled system */
+        VIR_INFO("ACRN hypervisor not available, disabling driver");
+        return 0;
```

Comment:
Can we goto cleanup here as well? virMutexInit() was already called previously so it'd be more graceful to destroy it. We still need to return 0 in this case.